src/main/java/org/alfresco/repo/solr/SOLRAdminClient.java

```diff
@@ -55,68 +61,42 @@
 import org.quartz.SchedulerException;
 import org.quartz.Trigger;
 import org.quartz.TriggerBuilder;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanFactory;
 import org.springframework.beans.factory.DisposableBean;
 import org.springframework.context.ApplicationEventPublisher;
 import org.springframework.context.ApplicationEventPublisherAware;
-
 /**
  * Provides an interface to the Solr admin APIs, used by the Alfresco Enterprise JMX layer.
  * Also tracks whether Solr is available, sending Spring events when its availability changes.
  * 
  * @since 4.0
  *
  */
-public class SOLRAdminClient implements ApplicationEventPublisherAware, DisposableBean
+public class SOLRAdminClient extends AbstractSolrAdminHTTPClient
+        implements ApplicationEventPublisherAware, DisposableBean, SolrAdminClient 
 {
-	private String solrHost;
-	private int solrPort;
-	private int solrSSLPort;
-	private String solrUrl;
-	private String solrUser;
-	private String solrPassword;
+
 	private String solrPingCronExpression;
 	private String baseUrl;
-	private CommonsHttpSolrServer server;
-	private int solrConnectTimeout = 30000; // ms
 
 	private ApplicationEventPublisher applicationEventPublisher;
 	private SolrTracker solrTracker;
 	
-	private HttpClientFactory httpClientFactory;
     private Scheduler scheduler;
+    
+    private List<SolrStoreMapping> storeMappings;
+    
+    private HashMap<StoreRef, SolrStoreMappingWrapper> mappingLookup = new HashMap<>();

```

Comment:
We probably don't need to define this as a HashMap (e.g. will Map do?)

Reply:
SolrClientUtil.extractMapping() is defined to accept a HashMap, so I'm using a HashMap to avoid casting on invocations.